frontend/context/StakingContractDetailsProvider.tsx

```diff
@@ -0,0 +1,143 @@
+import { useQueries } from '@tanstack/react-query';
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useCallback,
+  useContext,
+  useState,
+} from 'react';
+
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
+import { StakingProgramId } from '@/enums/StakingProgram';
+import { useAgent } from '@/hooks/useAgent';
+import { useChainId } from '@/hooks/useChainId';
+import { useStakingContractDetailsByStakingProgram } from '@/hooks/useStakingContractDetails';
+import { StakingContractDetails } from '@/types/Autonolas';
+
+import {
+  INITIAL_DEFAULT_STAKING_PROGRAM_ID,
+  StakingProgramContext,
+} from './StakingProgramProvider';
+
+/**
+ * hook to get all staking contract details
+ */
+const useAllStakingContractDetails = () => {
+  const stakingPrograms = Object.values([INITIAL_DEFAULT_STAKING_PROGRAM_ID]);
+  const chainId = useChainId();
+  const currentAgent = useAgent();
+
+  const queryResults = useQueries({
+    queries: stakingPrograms.map((programId) => ({
+      queryKey: REACT_QUERY_KEYS.ALL_STAKING_CONTRACT_DETAILS(
+        chainId,
+        programId,
+      ),
+      queryFn: async () =>
+        await currentAgent.serviceApi.getStakingContractDetailsByStakingProgram(
```

Comment:
there's gonna be a lot of multicalls, right, one big multicall per each Perl related contract. I wonder, if we can update the subgraph to include all staking contracts constants and do a one call something like `contracts`?
just a thing for future consideration

Reply:
definitely a good idea, especially if we can support multichain in the same request
though will need to hold on this until we migrate from current graph hosting to a more solid production stable one

Reply:
> can support multichain in the same request
no, at least not by using the graph studio unfortunately. should be one api endpoint per chain